application/comit_node/src/comit_client/bam.rs

```diff
@@ -150,69 +151,141 @@ fn to_json_value(
     serde_json::to_value(header).expect("converting bam-header to json must not fail")
 }
 
+#[derive(Debug)]
+enum ConnectionState {
```

Comment:
:wrench: I think naming those fields in the enum variants would greatly aid readability.